float64[30] position
